urllib3/util/selectors.py

```diff
@@ -50,60 +52,75 @@ def _fileobj_to_fd(fileobj):
     return fd
 
 
-def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
-    """ Wrapper function for syscalls that could fail due to EINTR.
-    All functions should be retried if there is time left in the timeout
-    in accordance with PEP 475. """
-    timeout = kwargs.get("timeout", None)
-    if timeout is None:
-        expires = None
-        recalc_timeout = False
-    else:
-        timeout = float(timeout)
-        if timeout < 0.0:  # Timeout less than 0 treated as no timeout.
-            expires = None
-        else:
-            expires = monotonic() + timeout
-
-    args = list(args)
-    if recalc_timeout and "timeout" not in kwargs:
-        raise ValueError(
-            "Timeout must be in args or kwargs to be recalculated")
-
-    result = _SYSCALL_SENTINEL
-    while result is _SYSCALL_SENTINEL:
+# Determine which function to use to wrap system calls because Python 3.5+
+# already handles the case when system calls are interrupted.
+if sys.version_info >= (3, 5):
+    def _syscall_wrapper(func, _, *args, **kwargs):
+        """ This is the short-circuit version of the below logic
+        because in Python 3.5+ all system calls automatically restart
+        and recalculate their timeouts. """
         try:
-            result = func(*args, **kwargs)
-        # OSError is thrown by select.select
-        # IOError is thrown by select.epoll.poll
-        # select.error is thrown by select.poll.poll
-        # Aren't we thankful for Python 3.x rework for exceptions?
+            return func(*args, **kwargs)
         except (OSError, IOError, select.error) as e:
-            # select.error wasn't a subclass of OSError in the past.
             errcode = None
             if hasattr(e, "errno"):
                 errcode = e.errno
             elif hasattr(e, "args"):
                 errcode = e.args[0]
```

Comment:
So it looks like the `.args` attribute is only an attribute of `select.error` before Python 3.3 ([PEP 3151](https://www.python.org/dev/peps/pep-3151)) and is just an alias of `OSError` in 3.3+ meaning it only has an `.errno` attribute. I'll submit a commit removing the dead code.